src/TimeZones.jl

```diff
@@ -4,10 +4,12 @@ using Dates
 using Printf
 using Serialization
 using Unicode
-import Dates: TimeZone
+
+import Dates: TimeZone, UTC
 
 export TimeZone, @tz_str, istimezone, FixedTimeZone, VariableTimeZone, ZonedDateTime,
-    DateTime, TimeError, AmbiguousTimeError, NonExistentTimeError, UnhandledTimeError,
+    DateTime, Date, Time, UTC, Local, TimeError, AmbiguousTimeError, NonExistentTimeError,
```

Comment:
I can do some organization. What do you think of grouping by the file in which these are defined (and having a comment indicating the file)

Reply:
This might be too granular, but:

```
export @tz_str
export AmbiguousTimeError, NonExistentTimeError, TimeError
export Date, DateTime, Local, Time, TimeZone, UTC, VariableTimeZone, ZonedDateTime
export isdatetime
```

I think it's less about what the groupings are but more that there is some order to them.

Reply:
I'll note that there is an ordering to this but it's not as obvious as I'd like. A couple of key things I would like to communicate as part of the exports:
- Quite a few of these are re-exports from `Dates` which are just here in case someone only does `using TimeZones`
- There are some types that are closely related like `UTC` and `Local` which I think should be sequential

I'll post an revised version in a sec

Reply:
What I'm currently thinking:
```julia
export
    TimeZone, FixedTimeZone, VariableTimeZone, @tz_str, istimezone, localzone
    UTC, Local,
    ZonedDateTime, timezone,

    # Exceptions
    TimeError, AmbiguousTimeError, NonExistentTimeError, UnhandledTimeError,

    # Discovery tools
    timezone_names, all_timezones, timezones_from_abbr, timezone_abbrs,
    next_transition_instant, show_next_transition,

    # Extended functionality from Dates. Note: Some of these exports have already been done
    # but are listed here for completeness.
    TimeZone, UTC,
    DateTime, Date, Time,
    now, today,
    year, month, week, day, dayofmonth, hour, minute, second, millisecond
    yearmonthday, yearmonth, monthday,
    firstdayofyear, firstdayofquarter, firstdayofmonth, firstdayofweek,
    lastdayofyear, lastdayofquarter, lastdayofmonth, lastdayofweek
```
I'm unsure on what is better at this point and possibly this kind of logical grouping should just be in the documentation? In any case I think this could be made into an issue and not hold up this PR.

Reply:
Agreed, this isn't a dealbreaker for the PR going through and can be discussed afterwards.

Side-note; does exporting functionality multiple times have any consequences to it?

Reply:
> does exporting functionality multiple times have any consequences to it?

No it shouldn't